services/src/main/java/org/keycloak/services/resources/admin/AuthenticationManagementResource.java

```diff
@@ -599,7 +618,34 @@ public void updateExecutions(@PathParam("flowAlias") String flowAlias, Authentic
             model.setRequirement(AuthenticationExecutionModel.Requirement.valueOf(rep.getRequirement()));
             realm.updateAuthenticatorExecution(model);
             adminEvent.operation(OperationType.UPDATE).resource(ResourceType.AUTH_EXECUTION).resourcePath(session.getContext().getUri()).representation(rep).success();
+            return Response.accepted(flow).build();
         }
+
+        AuthenticationFlowModel checkFlow = realm.getFlowByAlias(rep.getDisplayName());
+
+        //check if a flow with the same name already exist
+        if (checkFlow != null) {
+            //if only the description changed, the IDs of old and new are going to be the same
+            if (!checkFlow.getDescription().equals(rep.getDescription()) && checkFlow.getId().equals(rep.getFlowId())) {
+                checkFlow.setDescription(rep.getDescription());
+                realm.updateAuthenticationFlow(checkFlow);
+            } else {
+                //Flow with that name already exists
+                return ErrorResponse.exists("Flow alias name already exists");
+            }
+        } else {
+            //if the name changed
+            checkFlow = realm.getAuthenticationFlowById(rep.getFlowId());
+            checkFlow.setAlias(rep.getDisplayName());
```

Comment:
It seems the NPE can happen on this line. This is probably the reason of some more failing tests. Probably some null check is needed here?

Reply:
Fixed by throwing NotFoundException. Is that ok?

Reply:
It seems it is not ok. There are still failing tests in travis (for example RequiredActionTotpSetupTest) and I think that the reason is exactly this.

The execution doesn't necessarily need to have "flowId" filled. The AuthenticationExecutions, which point to the Authenticator implementations, don't have the "flowId" filled at all. I think you can just return from the method in case that checkFlow doesn't exists. Maybe something like this:
````
            if (checkFlow == null) {
                return;
            }
````

Reply:
I've added a check if its a flow or an execution that is being updated. Let's see the tests results

Reply:
I've done a rework of the endpoints and left the old code in the comments. Please let me know what you think. If the new approach is better than the previous one. The tests have passed. If you agree let me delete the commented code afterwards to have it all clean.